server/src/main/java/com/objectcomputing/checkins/services/checkins/CheckInController.java

```diff
@@ -95,8 +131,15 @@
      */
     @Get("/{id}")
     public CheckIn readCheckIn(@NotNull UUID id){
-        return checkInservices.read(id);
-    }
 
+        MemberProfile currentUser = currentUserServices.currentUserDetails();
+        Boolean isAdmin = securityService.hasRole(RoleType.Constants.ADMIN_ROLE);
+        CheckIn checkin = checkInservices.read(id);
 
+        if(currentUser.getUuid().equals(checkin.getTeamMemberId()) || currentUser.getUuid().equals(checkin.getPdlId()) || isAdmin) {
```

Comment:
Same here. Expensive validation

Reply:
We need to allow the current PDL as well. There needs to be a case added for this.

Reply:
currentUser.getUuid().equals(checkin.getPdlId()) refers to current PDL.